vividus-engine/src/main/java/org/vividus/log/LoggingTableTransformerMonitor.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.log;
+
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.jbehave.core.model.ExamplesTable.TableProperties;
+import org.jbehave.core.model.TableParsers;
+import org.jbehave.core.model.TableTransformerMonitor;
+import org.jbehave.core.model.TableTransformers.Formatting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LoggingTableTransformerMonitor implements TableTransformerMonitor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(LoggingTableTransformerMonitor.class);
+
+    private static final Formatting FORMATTING_TRANSFORMER = new Formatting();
+    private static final int MAX_ROWS_LOGGING = 50;
+
+    private final TableParsers tableParsers;
+
+    public LoggingTableTransformerMonitor(TableParsers tableParsers)
+    {
+        this.tableParsers = tableParsers;
+    }
+
+    @Override
+    public void beforeTransformerApplying(String transformerName, TableProperties properties, String inputTable)
+    {
+        String loggerFormat = String
+                .format("Applying table transformer: {}%nTabular parameters: {}%nInput table body:%n{}");
+        logTransformerInfo(transformerName, properties, inputTable, loggerFormat);
+    }
+
+    @Override
+    public void afterTransformerApplying(String transformerName, TableProperties properties, String outputTable)
+    {
+        String loggerFormat = String
+                .format("Applying table transformer: {}%nTabular parameters: {}%nOutput (transformed) table:%n{}");
+        logTransformerInfo(transformerName, properties, outputTable, loggerFormat);
+    }
+
+    private void logTransformerInfo(String transformerName, TableProperties tableProperties, String table,
+            String loggerFormat)
+    {
+        String propertiesForLogging = tableProperties.getPropertiesAsString().trim();
```

Comment:
I would expect the properties logged in the following format:
```
{headerSeparator=..., }
|k1|k2|
...
```
and they should be logged only if there values which differ from default ones

Reply:
Did you mean add braces?
"getPropertiesAsString" returns by default only user properties
But I don't think that we must skip property section at all, because transformer is also a property (which can have its own properties)

Reply:
@valfirst 
Since we don't have a strict separation between the properties of the table and the properties of the transformer I would not expect that we may not have properties
![tableProps](https://user-images.githubusercontent.com/58571912/204223357-661d5c9f-c922-47cb-af90-1397bf16dcfd.png)